### DISCUSSION TITLE
Is there a way to convert a xarray to numpy array?

### DISCUSSION THREAD

**Discussion by YosmelyBermudez** [pydata/xarray#5784] [Q&A] [answered]
Hi!I am generating a code to work seismic data and it happens that the segysak library only works with xarray and I need to pass it to numpy array. I have tried with commands like to_numpy or passing dask dataframe to bring it to numpy array and nothing! HELP!These were the lines of code I used, the xarray is seisnc_vol:
original_cube = np.array(seisnc_vol)
original_cube = np.ascontiguousarray(original_cube, 'float32')

print("Original cube dimension:", original_cube.shape)

**Comment by keewis** [comment, marked as answer]
what type does `seisnc_vol` have? If it's a `DataArray` object, the conversion should be as simple as accessing `seisnc_vol.values` or calling `np.asarray(seisnc_vol)` (since you already tried `np.array(seisnc_vol)` I'm guessing it's not). If it's a `Dataset` object, there are a few more steps involved.

Can you post the output of `print(seisnc_vol)`? That might make it easier to help you.

**Reply by YosmelyBermudez** [reply]
you're right! it is a DataArray object and I can work it with the value. thanks a lot!
